src/views/tripPlannerWrapper.jsx

```diff
@@ -0,0 +1,225 @@
+      <p> - No emergency contact provided.</p>
+    );
+    if(d.overdueTime&&(((d.overdueTime.getTime()-d.returnTime.getTime())/(1000*60*60))<2)) warnings.push(
+      <p> - Overdue time is within 2hrs of return. Consider allowing more time before action is required.</p>
+    );
+
+    // If no warnings, continue. Else, show modal.
+    if(warnings.length===0){
+      saveTrip()
+    } else {
+      app.setModal(
+        <WarningModal
+          warnings={warnings}
+          onContinue={()=>{app.hideModal(); saveTrip()}}
+          hideModal={()=>{app.hideModal(); returnError("")}}
+        />
+      )
+    }
+  }
+
+  function saveTrip(){
+    let newTrip = trip ? false:true
+    let d = dataToSave
+    let contactIds = []
+    for(let i in d.contacts){
+      contactIds.push(d.contacts[i].id)
+    }
+
+    // Follow format of backend serializer class
+    let tripData={
+      name: d.name,
+      departs: d.departTime.toISOString(),
+      returns: d.returnTime.toISOString(),
+      description: d.description,
+      points: JSON.stringify(d.points),
+      paths: JSON.stringify(d.paths),
+      contacts: contactIds,
+      overdue: d.overdueTime ? d.overdueTime.toISOString() : null,
+      instructions: d.overdueTime ? d.overdueInstructions : "",
+    }
+    console.log("Data is about to be saved: ", tripData)
+
+    apiFetch({
+      url: newTrip ? urls.MY_TRIPS : `${urls.TRIP}/${trip.id}/`,
+      method: newTrip ? "POST" : "PATCH",
+      data: tripData,
+      onSuccess: handleSuccess,
+      onFailure: handlePostFailure,
+    })
+  }
+
+  function handleSuccess(newTrip){
+    console.log("TRIP SAVED!")
+    console.log(newTrip)
+    app.refresh()
+    console.log("Redirect to home")
+    setRedirect(urls.HOME)
+  }
+
+  function handlePostFailure(error){
+    console.log(error)
+    returnError(`Unable to save ${trip ? "changes":"new trip"}.`)
+  }
+
+  function returnError(message){
+    setErrorMessage(message)
+    setPending(false)
+    setDataToSave(null)
+    return false
+  }
+
+  function confirmDelete(){
+    setPending(true)
+    app.setModal(
+      <WarningModal 
+        body={<div>Are you sure you want to delete this trip?</div>} 
+        onContinue={()=>{app.hideModal(); deleteTrip()}}
+        hideModal={()=>{app.hideModal(); returnError("")}}
+      />)
+  }
+
+  function deleteTrip(){
+    console.log("DELETE")
+    apiFetch({
+      url: `${urls.TRIP}/${trip.id}/`,
+      method: "DELETE",
+      onSuccess:()=>{
+        app.refresh()
+        console.log("Redirect to home")
+        setRedirect(urls.HOME)
+      },
+      onFailure: handlePostFailure,
+    })
+  }
+
+  // Redirect to home on save or delete
+  if(redirect) return <Redirect push={true} to={redirect} />
+   
+  if(!app.loggedIn){
+    return (app.loginPending ?
+      <LoadingScreen />
+      :
+      <Redirect push={true} to={urls.HOME} />
+    ) 
+  }
+
+  return(
+    <TripPlannerForm
+      app={app}
+      user={user}
+      trip={trip}
```

Comment:
It ALMOST seems like `TripPlannerWrapper` is the perfect place to determine if a trip already exists or not, then pass the default form values as props to `TripPlannerForm`... 🤔

Reply:
This is good, I think that would work. What do you think about the existing devision of logic between the wrapper and form? I tried to separate all the API calls and validation from the rendering and basic state changes. I feel like I could abstract away a lot of the CRUD calls to the backend, because a lot of those are duplicated around the project.

Reply:
@mccutch Ya the general idea of seperating the fetching, defining some handlers is good. We could probably look at how these handler functions are structured, and really start getting a better sense of what belongs here vs what belongs somewhere else. Id superate TripEdit into a different component also just to not have it staring at me lol. 

A common pattern is to save React component files with a capital first letter, and functions that DONT returns jsx in regular camelcase.